python/pyarrow/_csv.pyx

```diff
@@ -299,6 +352,48 @@ cdef class ParseOptions(_Weakrefable):
         parsing (because of a mismatching number of columns).
         It should accept a single InvalidRow argument and return either
         "skip" or "error" depending on the desired outcome.
+
+    Examples
+    --------
+
+    Defining an example file from bytes object:
+
+    >>> import io
+    >>> s = b'''animals;n_legs;entry
+    ... Flamingo;2;2022-03-01
+    ... # Comment here:
+    ... Horse;4;2022-03-02
+    ... Brittle stars;5;2022-03-03
+    ... Centipede;100;2022-03-04'''
+
+    Read the data from a file skipping rows with comments
+    and defining the delimiter:
+
+    >>> from pyarrow import csv
+
+    >>> class InvalidRowHandler:
+    ...     def __init__(self, result):
+    ...         self.result = result
+    ...     def __call__(self, row):
+    ...         if row.text.startswith("# "):
+    ...             return self.result
+    ...         else:
+    ...             return 'error'
+    ...
+    >>> skip_handler = InvalidRowHandler('skip')
```

Comment:
We used such a class in the tests (to make it more flexible, and to test what was passed), but for this example here, it could maybe also be a simpler function? Something like:

```
def skip_comment(row):
    if row.text.startswith("# "):
        return "skip"
    return "error"
```

(didn't test if it works)

Reply:
Yes, that would be better. Will try it out, thanks!